### PR TITLE
Zero-indexed parts, for sure

### LINKED ISSUE #56: Clarify valid values for --part parameter
I've been using mapbox-tile-copy to process some very large .shp files, using the option to process them in parts.

However, after looking at the documentation I'm still not quite sure about if the --part parameter is zero indexed (e.g. if --parts=10 then valid values for part is 0..9) or if it instead should start at 1 (e.g. if --parts==10 then valid values for part is 1..10).

Trial and error has not really helped me either since I've not reliably been able to process either part 0 or 10 (in case of 10 parts). Sometimes part=0 works (it doesn't contain a lot of tiles though) but most of the time it will just time out (and trying part=10 always seems to time out).

`Error: Copy operation timed out
    at null.<anonymous> (/usr/lib/node_modules/mapbox-tile-copy/node_modules/tilelive/lib/tilelive.js:390:22)
    at wrapper (timers.js:261:14)
    at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)`

It would be great if the documentation could make this a bit more clear :). Thanks and a huge thank you for open sourcing this awesome project!

### PR DESCRIPTION
**Pull request by GretaCB**
Add test to ensure a zero-indexed single-part copy job includes all tiles and update docs to include clarity around `--parts` operation.

cc @rclark @mapsam 
Fixes #56

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -37,14 +37,24 @@ Copy tiles from one S3 location to another via tilejson describing the source:
...
-Render image tiles from vector tiles, using custom fonts from a location on your computer:
+The `--part` operation is explicitly _zero-indexed_ because this gives tilelive's stream processors a predictable way to segment tiles per part. For example, the following will distribute all tiles among a single part. So all tiles will be rendered by this single part:
 ```sh
-$ MapboxTileCopyFonts=/path/to/font/dir mapbox-tile-copy ~/style.tm2z s3://my-bucket/pngs/{z}/{x}/{y}
+$ mapbox-tile-copy ~/data/my-tiles.mbtiles s3://my-bucket/parallel/{z}/{x}/{y} --part 0 --parts 1
+```
+
+The following example will distribute tiles to the second part out of 5 total parts:
+```sh
+$ mapbox-tile-copy ~/data/my-tiles.mbtiles s3://my-bucket/parallel/{z}/{x}/{y} --part 1 --parts 4
```
Comment (rclark) [inline readme.md]: \o/ 🎉 🎆
